fieldservice_activity/tests/test_fsm_activity.py

```diff
@@ -67,3 +68,21 @@ def get_activity_vals(self, name, required, ref, order_id):
             'ref': ref,
             'fsm_order_id': order_id
         }
+
+    def test_onchange_template_id(self):
+        # Create a Template
+        self.template = self.template_obj.create(
+            {'name': 'Demo template',
+             'temp_activity_ids': [(0, 0, {
+                 'name': "Activity new",
+                 'required': True,
+                 'ref': 'Ref new',
+                 'state': 'todo',
+             })]})
+        # Create an Order
+        self.fso = self.Order.create({
+            'location_id': self.test_location.id,
+            'template_id': self.template.id,
+        })
+        # Test _onchange_template_id()
+        self.fso._onchange_template_id()
```

Comment:
@murtuzasaleh Can you check that the order has 1 activity and the activity is different from the one related to the template?